meson.build

```diff
@@ -1,86 +1,119 @@
-project('Little-CMS', 'c',
+project(
+  'Little-CMS',
+  'c',
   version: '2.14',
-  meson_version: '>=0.48.0',
+  meson_version: '>=0.49.0',
 )
 
-library_version = '2.0.13'
+version_components = meson.project_version().split('.')
+
+library_version = '@0@.0.@1@'.format(
+  version_components.get(0, 0),
+  version_components.get(1, 0),
+)
+
+version_cfg = configuration_data()
+version_cfg.set('LCMS2_VERSION_MAJOR', version_components.get(0, 0))
+version_cfg.set('LCMS2_VERSION_MINOR', version_components.get(1, 0))
+version_cfg.set('LCMS2_VERSION_MICRO', version_components.get(2, 0))
+version_cfg.set_quoted('LCMS2_VERSION', meson.project_version())
 
 cc = meson.get_compiler('c')
 
-lcms2_srcs = [
-  'src/cmsalpha.c',
-  'src/cmscgats.c',
-  'src/cmserr.c',
-  'src/cmsgmt.c',
-  'src/cmsintrp.c',
-  'src/cmsio1.c',
-  'src/cmsmd5.c',
-  'src/cmsnamed.c',
-  'src/cmspack.c',
-  'src/cmsplugin.c',
-  'src/cmssamp.c',
-  'src/cmstypes.c',
-  'src/cmswtpnt.c',
-  'src/cmscam02.c',
-  'src/cmscnvrt.c',
-  'src/cmsgamma.c',
-  'src/cmshalf.c',
-  'src/cmsio0.c',
-  'src/cmslut.c',
-  'src/cmsmtrx.c',
-  'src/cmsopt.c',
-  'src/cmspcs.c',
-  'src/cmsps2.c',
-  'src/cmssm.c',
-  'src/cmsvirt.c',
-  'src/cmsxform.c',
-]
-
-install_headers([
-  'include/lcms2.h',
-  'include/lcms2_plugin.h'
-])
-
-inc_dirs = include_directories('include')
-
-deps = [
-  cc.find_library('m', required: false),
-  dependency('threads'),
-]
-
-cargs = ['-DHasTHREADS=1']
+cargs = []
+
+if cc.has_function_attribute('visibility:hidden')
+  cargs += '-DHAVE_FUNC_ATTRIBUTE_VISIBILITY=1'
+endif
+
+if host_machine.endian() == 'big'
+  cargs = '-DWORDS_BIGENDIAN=1'
+endif
+
 
 # Check for threadsafe variants of gmtime
-if cc.has_function('gmtime_r', prefix: '#include <time.h>')
+# MinGW needs _POSIX_C_SOURCE or _POSIX_THREAD_SAFE_FUNCTIONS defined
+# to make gmtime_r available
+if host_machine.system() == 'windows' and cc.get_argument_syntax() != 'msvc'
+  cargs += ['-D_POSIX_C_SOURCE=199503L']
+endif
+
+if cc.has_header_symbol(
+  'time.h',
+  'gmtime_r',
+  args: cargs,
+)
   cargs += '-DHAVE_GMTIME_R=1'
-elif cc.has_function('gmtime_s', prefix: '#include <time.h>')
-  cargs += '-DHAVE_GMTIME_S=1'
+elif cc.has_header_symbol('time.h', 'gmtime_s')
+  if cc.links(
+    '''
+    #include <time.h>
+
+    int main() {
+      time_t t;
+      struct tm m;
+      gmtime_s(&m, &t);
+      return 0;
+    }
+    ''',
+    name: 'gmtime_s can be used',
+  )
+    cargs += '-DHAVE_GMTIME_S=1'
+  endif
 endif
 
-if cc.has_function_attribute('visibility:hidden')
-  cargs +='-DHAVE_FUNC_ATTRIBUTE_VISIBILITY=1'
+jpeg_dep = dependency('libjpeg', required: get_option('jpeg'))
+
+if jpeg_dep.found()
+  cargs += ['-DHasJPEG=1']
```

Comment:
I grepped the sources and AFAICT it's an inefficiency of the configure.ac / Makefile.am that `HasJPEG`, `HasTIFF`, or `HasZLIB` are ever defined as compiler macros.

#348

Reply:
Alright, removed.